src/test/java/com/iota/iri/conf/ConfigFactoryTest.java

```diff
@@ -0,0 +1,170 @@
+        assertTrue("Expected iotaConfig as instance of TestnetConfig.", iotaConfig instanceof TestnetConfig);
+        assertTrue("Expected iotaConfig as Testnet.", iotaConfig.isTestnet());
+    }
+
+    /**
+     * Creates and validates a Testnet {@link IotaConfig} with <code>TESTNET=true</code> in config file and
+     * <code>testnet: true</code> as method parameter for {@link ConfigFactory#createFromFile(File, boolean)}.
+     * @throws IOException when config file not found.
+     */
+    @Test
+    public void createFromFileTestnetWithTestnetTrueAndTrue() throws IOException {
+        // lets assume in our configFile is TESTNET=true
+        File configFile = createTestnetConfigFile("true");
+
+        // but the parameter is set to testnet=true
+        IotaConfig iotaConfig = ConfigFactory.createFromFile(configFile, true);
+        assertTrue("Expected iotaConfig as instance of TestnetConfig.", iotaConfig instanceof TestnetConfig);
+        assertTrue("Expected iotaConfig as Testnet.", iotaConfig.isTestnet());
+    }
+
+    /**
+     * Creates and validates a Testnet {@link IotaConfig} with <code>TESTNET=false</code> in config file and
+     * <code>testnet: true</code> as method parameter for {@link ConfigFactory#createFromFile(File, boolean)}.
+     * @throws IOException when config file not found.
+     */
+    @Test
+    public void createFromFileTestnetWithTestnetFalseAndTrue() throws IOException {
+        // lets assume in our configFile is TESTNET=false
+        File configFile = createTestnetConfigFile("false");
+
+        // but the parameter is set to testnet=true
+        IotaConfig iotaConfig = ConfigFactory.createFromFile(configFile, true);
+        assertTrue("Expected iotaConfig as instance of TestnetConfig.", iotaConfig instanceof TestnetConfig);
+        assertTrue("Expected iotaConfig as Testnet.", iotaConfig.isTestnet());
+    }
+
+    /**
+     * Creates and validates a Mainnet {@link IotaConfig} with <code>TESTNET=false</code> in config file and
+     * <code>testnet: false</code> as method parameter for {@link ConfigFactory#createFromFile(File, boolean)}.
+     * @throws IOException when config file not found.
+     */
+    @Test
+    public void createFromFileTestnetWithTestnetFalseAndFalse() throws IOException {
+        // lets assume in our configFile is TESTNET=false
+        File configFile = createTestnetConfigFile("false");
+
+        // but the parameter is set to testnet=true
+        IotaConfig iotaConfig = ConfigFactory.createFromFile(configFile, false);
+        assertTrue("Expected iotaConfig as instance of MainnetConfig.", iotaConfig instanceof MainnetConfig);
+        assertFalse("Expected iotaConfig as Mainnet.", iotaConfig.isTestnet());
+    }
+
+    /**
+     * Test if leading and trailing spaces are trimmed from string in properties file.
+     * @throws IOException when config file not found.
+     */
+    @Test
+    public void createFromFileTestnetWithTrailingSpaces() throws IOException {
```

Comment:
Perhaps we should a nice separation where `ConfigTest` tests parsing and `ConfigFactoryTest` tests creation?